src/cpp/modules/manualEigen/ba_d.cpp

```diff
@@ -0,0 +1,387 @@
+    distort_proj_d[0 * 2 + 0] = distort_proj_d_coef * proj[0] * proj[0] + L;
+    distort_proj_d[1 * 2 + 0] = distort_proj_d_coef * proj[0] * proj[1];
+    distort_proj_d[0 * 2 + 1] = distort_proj_d_coef * proj[1] * proj[0];
+    distort_proj_d[1 * 2 + 1] = distort_proj_d_coef * proj[1] * proj[1] + L;
+    distort_rad_d[0 * 2 + 0] = rsq * proj[0];
+    distort_rad_d[0 * 2 + 1] = rsq * proj[1];
+    distort_rad_d[1 * 2 + 0] = rsq * distort_rad_d[0 * 2 + 0];
+    distort_rad_d[1 * 2 + 1] = rsq * distort_rad_d[0 * 2 + 1];
+    proj[0] *= L;
+    proj[1] *= L;
+}
+
+// Arguments:
+// cam - double[BA_NCAMPARAMS]
+// X - double[3]
+// proj - double[2]
+// J - 2 x (BA_NCAMPARAMS+3+1) in column major
+void project_d(
+    const double* const cam,
+    const double* const X,
+    double* proj,
+    double* J)
+{
+    double Xo[3], Xcam[3];
+    const double* const rot = &cam[BA_ROT_IDX];
+    const double* const C = &cam[BA_C_IDX];
+    double f = cam[BA_F_IDX];
+    const double* const x0 = &cam[BA_X0_IDX];
+    const double* const rad = &cam[BA_RAD_IDX];
+
+    double* Jrot = &J[2 * BA_ROT_IDX];
+    double* JC = &J[2 * BA_C_IDX];
+    double* Jf = &J[2 * BA_F_IDX];
+    double* Jx0 = &J[2 * BA_X0_IDX];
+    double* Jrad = &J[2 * BA_RAD_IDX];
+    double* JX = &J[2 * BA_NCAMPARAMS];
+
+    subtract(3, X, C, Xo);
+
+    double rodri_rot_d[9], rodri_Xo_d[9];
+    rodrigues_rotate_point_d(rot, Xo, Xcam, rodri_rot_d,
+        rodri_Xo_d);
+
+    p2e(Xcam, proj);
+
+    double distort_proj_d[4], distort_rad_d[4];
+    radial_distort_d(rad, proj, distort_proj_d, distort_rad_d);
+
+    double hnorm_right_col[2];
+    double tmp = 1. / (Xcam[2] * Xcam[2]);
+    hnorm_right_col[0] = -Xcam[0] * tmp;
+    hnorm_right_col[1] = -Xcam[1] * tmp;
+    double distort_hnorm_d[6];
+
+    for (int i = 0; i < 4; i++)
+        distort_hnorm_d[i] = distort_proj_d[i] / Xcam[2];
+    distort_hnorm_d[2 * 2 + 0] = distort_proj_d[0 * 2 + 0] * hnorm_right_col[0] + distort_proj_d[1 * 2 + 0] * hnorm_right_col[1];
+    distort_hnorm_d[2 * 2 + 1] = distort_proj_d[0 * 2 + 1] * hnorm_right_col[0] + distort_proj_d[1 * 2 + 1] * hnorm_right_col[1];
+
+    mat_mul(2, 3, 3, distort_hnorm_d, rodri_rot_d, Jrot);
+    for (int i = 0; i < 6; ++i)
+        Jrot[i] *= f;
+
+    mat_mul(2, 3, 3, distort_hnorm_d, rodri_Xo_d, JC);
+    for (int i = 0; i < 6; ++i)
+        JC[i] *= -f;
+    Jf[0] = proj[0];
+    Jf[1] = proj[1];
+
+    // Initializind Jx0 to identity
+    Jx0[0] = 1.;
+    Jx0[1] = 0.;
+    Jx0[2] = 0.;
+    Jx0[3] = 1.;
+    for (int i = 0; i < 4; ++i)
+        Jrad[i] = distort_rad_d[i] * f;
+    for (int i = 0; i < 6; ++i)
+        JX[i] = -JC[i];
+
+    proj[0] = proj[0] * f + x0[0];
+    proj[1] = proj[1] * f + x0[1];
+}
+
+// Arguments:
+// cam - double[BA_NCAMPARAMS]
+// X - double[3]
+// w, feat_x, feat_y - double
+// err - double[2]
+// J - 2 x (BA_NCAMPARAMS+3+1) in column major
+void compute_reproj_error_d(
```

Comment:
This function is a copy of one from manual/ba_d.cpp and it does not use Eigen in the computation. In fact, the code in this file that uses Eigen doesn't seem to be called from anywhere.